### ISSUE TITLE
Questions about "soft labels"

### ISSUE THREAD

**Issue by yihudiana3** [cliang1453/BOND#6]
https://github.com/cliang1453/BOND/blob/3651a92adedbd53bbc6210fade7f688b171d6434/run_self_training_ner.py#L240

Hello, a few questions came to mind when I read the code dealing with "soft labels", and I wonder if you could kindly help:

1. What's the difference between "label_mask" and "attention_mask" here? And since BERT forward doesn't take "label_mask" as input, including the keyword "label_mask" seems to cause an unexpected keyword argument error when inputs is passed to model

2. It seems to me that pred_labels is of shape (batch_size , sequence_length , num_labels), which is different than what is accepted by BERT labels, i.e. (batch_size, sequence_length)(thus potential source of size does not match error when passed to BERT forward). And accoding to the paper, losses associated with soft labels are calculate in a different way than BERT loss, but in the code, soft label loss is also calculated by passing inputs to BERT model, so I'm a bit confused.

It would be grateful if you could kindly help, and have a nice day!

**Comment by mouxiaofeng1981** [comment]
I have meet the same problem.

**Comment by cliang1453** [comment]
Hi, 

1. ''label_mask'' is a binary mask of selected tokens with high prediction confidence. We optimize losses only on such tokens (See Paper Section 3.2.2). Unfortunately, our codes only support RoBERTa, where we implement a new classification head in RobertaForTokenClassification_v2. To use BERT, You'll need to implement new classification head instead of using BertForTokenClassification. 
2. The soft label implementation is in https://github.com/cliang1453/BOND/blob/3651a92adedbd53bbc6210fade7f688b171d6434/modeling_roberta.py#L92. 

Hope this helps!
